### ISSUE TITLE
Make provider name resolution case insensitive

### ISSUE THREAD

**Issue by nyalldawson** [qgis/qgis3.0_api#109]
We currently have a gross mix of provider names in upper/lower case. E.g. "WFS" vs "wms". It's confusing even for experienced developers!

We should either unify the case of all the names (more porting work required for 4.0 plugin upgrades), or just make the name resolution case insensitive (possibly we could even do this before 4.0 -- comments welcome!).

I can't see that there'd ever be a valid use case for two different providers with the same name but different casing...

**Comment by NathanW2** [comment]
I would say it can be done now, can't see it breaking anything...

**Comment by m-kuhn** [comment]
Yes, do that now unifying capitalization in 4.0 sounds good.

**Comment by elpaso** [comment]
Yes, please, this bugged me a lot in the past (now I think I memorized the uppercase ones).
